class-two-factor-core.php

```diff
@@ -480,6 +483,23 @@ public static function filter_authenticate( $user ) {
 		return $user;
 	}
 
+	/**
+	 * Prevent login cookies being set on login for Two Factor users.
+	 *
+	 * @param  WP_User|WP_Error $user Valid WP_User only if the previous filters
+	 *                                have verified and confirmed the
+	 *                                authentication credentials.
+	 *
+	 * @return WP_User|WP_Error
+	 */
+	public static function filter_authenticate_block_cookies( $user ) {
+		if ( $user instanceof WP_User && self::is_user_using_two_factor( $user->ID ) && did_action( 'login_init' ) ) {
```

Comment:
Additionally, the `did_action( 'login_init' )` here is also for edge-case protection.

This allows for 3rd-party code authenticating the user through a non-wp-login.php page, yet using the `authenticate` filter to still call `wp_set_auth_cookie()`.

This is at odds with #385 but as long as this is noted for when/if that issue is resolved.

Reply:
Same opinion here about adding a comment.